workspace_api/views.py

```diff
@@ -113,6 +127,39 @@ async def create_workspace(
     return {"name": workspace_name}
 
 
+def create_bucket_secret(workspace_name: str, credentials: Dict[str, Any]) -> None:
+
+    logger.info(f"Creating secret for namespace {workspace_name}")
+    k8s_client.CoreV1Api().create_namespaced_secret(
+        namespace=workspace_name,
+        body=k8s_client.V1Secret(
+            metadata=k8s_client.V1ObjectMeta(
+                name=CONTAINER_REGISTRY_SECRET_NAME,
+            ),
+            data={
+                "username": base64.b64encode(credentials["workspace_name"].encode()).decode(),
+                "password": base64.b64encode(
+                    credentials["password"].encode()
+                ).decode(),
+            },
+        )
+    )
+
+
+async def wait_for_secret(
+        workspace_name: str, res: object , time_out: int, time_interval: int) -> V1Secret:
+
+    timer = 0
+    while res.status == 201:
+        await asyncio.sleep(time_interval)
+        timer += time_interval
+        if timer > time_out:
+            # TODO: raise a timeout exception
+            break
+        if res.status == 200:
+            return create_bucket_secret(workspace_name=workspace_name, credentials=res.body)
+
+
```

Comment:
Remove this function again.

inside the `install_workspace_phase2` in the `wait_for_namespace_secret` we make a check first if the secret already exists using `fetch_secret`, exiting ASAP.